### PR TITLE
Allow specifying multiple port ranges in sni.yaml

### PR DESCRIPTION
**Pull request by JosiahWI**
This changes sni.yaml parsing to allow multiple port ranges.

Before:
```yaml
sni:
- fqdn: example.com
  inbound_port_range: 443-446
```

After:
```yaml
sni:
- fqdn: example.com
  inbound_port_ranges:
  - 443-446
  - 8080
- fqdn: other.com
  inbound_port_ranges: 443
  ```
  
  Test cases and documentation have been updated to reflect the change, and parts of the parsing and AuTest have been refactored.
  
  TODO:
  
  - [x] Consider overlapping ranges (would probably indicate a mistake, but permitted for now as it doesn't do any harm)
  - [x] Double check behavior of invalid ranges where `min` is greater than `max` (already tested 😄)
  - [x] Consider allowing both a sequence or a scalar value for `inbound_port_ranges`. (implemented)

### CODE REVIEW

File: iocore/net/YamlSNIConfig.cc [hunk 1]
```diff
@@ -213,6 +213,38 @@ std::set<std::string> valid_sni_config_keys = {TS_fqdn,
...
 {
 template <> struct convert<YamlSNIConfig::Item> {
+  static ts::port_range_t
+  parse_single_inbound_port_range(Node const &node, swoc::TextView port_view)
+  {
+    auto min{port_view.split_prefix_at('-')};
+    if (!min) {
+      min = port_view;
+    }
+    auto const &max{port_view};
```
Comment (SolidWallOfCode) [inline iocore/net/YamlSNIConfig.cc:223]: Don't use references with views - they are essentially references already.

File: iocore/net/YamlSNIConfig.cc [hunk 2]
```diff
@@ -213,6 +213,38 @@ std::set<std::string> valid_sni_config_keys = {TS_fqdn,
...
+
+    swoc::TextView parsed_min;
+    long min_port{swoc::svtoi(min, &parsed_min)};
+    swoc::TextView parsed_max;
+    long max_port{swoc::svtoi(max, &parsed_max)};
```
Comment (SolidWallOfCode) [inline iocore/net/YamlSNIConfig.cc:228]: Use `auto max_port = ...`

File: iocore/net/YamlSNIConfig.cc (hunk 1 above)
Comment (JosiahWI) [reply iocore/net/YamlSNIConfig.cc:223]: Thanks for catching this. I've caught myself doing this in other places and am trying to develop the new habit.

File: iocore/net/YamlSNIConfig.cc (hunk 2 above)
Comment (JosiahWI) [reply iocore/net/YamlSNIConfig.cc:228]: What's the reason for using copy initialization over universal initialization in this case?

**Comment by JosiahWI** [comment]
[approve ci autest]

File: iocore/net/YamlSNIConfig.cc (hunk 2 above)
Comment (cmcfarlen) [reply iocore/net/YamlSNIConfig.cc]: There are plenty of style guides that recommend `=` over uniform initialization for trivial initialization like this, especially with `auto` where narrowing conversions are not a concern.  Style-wise, in ATS, this is certainly the norm.

**Review by SolidWallOfCode — APPROVED** [review]

**Pull request merged**